hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestWorkPreservingRMRestart.java

```diff
@@ -1651,6 +1651,94 @@ public void testUAMRecoveryOnRMWorkPreservingRestart() throws Exception {
     assertUnmanagedAMQueueMetrics(qm2, 1, 0, 0, 1);
   }
 
+  //   Test behavior of an app if two same name leaf queue with different queuePath
+  //   during work preserving rm restart with %specified mapping Placement Rule.
+  //   Test case does following:
+  //1. Submit an apps to queue root.p1.test.
+  //2. During the applications is running, restart the rm and
+  //   check whether the app submitted to the queue it was submitted initially.
+  //3. Verify that application running successfully.
+  @Test(timeout = 60000)
+  public void testQueueRecoveryOnRMWorkPreservingRestart() throws Exception {
+    CapacitySchedulerConfiguration csConf = new CapacitySchedulerConfiguration(conf);
```

Comment:
Since this test is CS related can you please add a condition to skip it if the chosen scheduler is not CapacityScheduler?

Like above:
`if (getSchedulerType() != SchedulerType.CAPACITY) {
      return;
    }`